glean-core/csharp/Glean/Dispatchers.cs

```diff
@@ -39,61 +39,44 @@ internal class Dispatchers
         private static int _queueInitialTasks = 1;
         /// <summary>
         /// When true, tasks will be queued and not ran until triggered by
-        /// calling FlushQueuedInitialTasks.
+        /// calling FlushQueuedInitialTasks.  This uses an int backing field
+        /// in order to take advantage of Interlocked.Exchange for thread
+        /// safety.
         /// </summary>
         internal static bool QueueInitialTasks
         {
             get => _queueInitialTasks == 1;
             set => Interlocked.Exchange(ref _queueInitialTasks, value ? 1 : 0);
         }
 
-        // Create a ConcurrentExclusiveSchedulerPair object.
-        // All API tasks will be executed on the exclusive part of the scheduler
-        // in order to ensure single threaded behavior
-        private static readonly
-            ConcurrentExclusiveSchedulerPair taskSchedulerPair =
-            new ConcurrentExclusiveSchedulerPair();
-
-        /// <summary>
-        /// This Queue holds the initial tasks that are launched before Glean is
-        /// initialized.
-        /// </summary>
-        internal static ConcurrentQueue<Action> taskQueue =
-            new ConcurrentQueue<Action>();
+        // Create a scheduler that uses a single thread
+        private static readonly LimitedConcurrencyLevelTaskScheduler lcts =
+            new LimitedConcurrencyLevelTaskScheduler(1);
 
-        /// <summary>
-        /// The number of tasks added to the queue beyond the MaxQueueSize.
-        /// </summary>
-        private static int OverflowCount { get; set; } = 0;
+        // Create a new TaskFactory and pass it the scheduler and cancellation
```

Comment:
We're not passing a cancellation token anymore!

Reply:
Yeah, that was part of the problem, I think.  If you are okay with it I'll file a follow-up bug to implement that part back (if we even need it).

Reply:
No, I believe it's fine to not have that.